syn/routes/api/v1/utils.py

```diff
@@ -70,3 +72,34 @@ def tokens():
                 res[chain][token].update({'cgid': cgid})
 
     return res
+
+
+@utils_bp.route('/token_price', methods=['GET'])
+@cache.cached()
+def token_price():
+    chain_id = request.args.get('token', None)
+    token = request.args.get('token', None)
+    date = request.args.get('date', None)
+
+    # validate chain id
+    if not verify.isdigit(chain_id) or (chain_id := int(chain_id) not in CHAINS_REVERSED):
+        return jsonify({'error': 'invalid chain id'}), 400
+    chain_name = CHAINS_REVERSED[chain_id]
+
+    # validate token address
+    if token not in ADDRESS_TO_CGID[chain_name]:
+        return jsonify({'error': 'token for chain not supported'}), 400
+
+    # validate date format if entered
+    if date:
+        try:
+            datetime.strptime(date, '%d-%m-%Y')
```

Comment:
use `datetime.fromisoformat()`

Reply:
This works for yyyy-mm-dd, not dd-mm-yyyy which is what coingecko supports. I think I should accept an ISO date and then internally format it to coingecko date

Reply:
Oh I see `dateutil.parser.parse` in `get_historic_price` fixes this issue. Alright